axon-server-connector/src/main/java/org/axonframework/axonserver/connector/query/AxonServerQueryBus.java

```diff
@@ -608,6 +644,11 @@ public Builder localSegment(QueryBus localSegment) {
             return this;
         }
 
+        public Builder localSegmentShortCut(boolean localSegmentShortCut) {
```

Comment:
Nice, thanks for adding this!
I have a small suggestion on this, though.

In other infrastructure components with such a flag, we picked the sensible default and named the method such that it describes what it switches to.

What I think would be a good default for this feature, is `false`. Mainly because if we set it to `true,` it changes behavior for everybody. So, it is not so much a breaking change in the API but a breaking change in behavior.
Hence why `false` would be the best option for this I think.

Knowing the default, we can thus decide on the name for this method. As it would switch it from `false` to `true`, I think calling it `enabledLocalSegmentShortCut` (without any parameters) would be nice.

But, I am curious to hear what you think about this, @rsobies!

```suggestion
        public Builder enabledLocalSegmentShortCut() {
```

Reply:
>What I think would be a good default for this feature, is false. Mainly because if we set it to true, it changes behavior for everybody. So, it is not so much a breaking change in the API but a breaking change in behavior.
Hence why false would be the best option for this I think.

I understand your point, but on other hand this shortcut will be a kind of optimalization so I suspect that most users maybe even all, would prefer to set that value to true, don't you agree?

Reply:
Although I also understand your point of view, @rsobies, I have also made this assumption in the past with extremely differing results. 
Hence I have decided to be careful on introducing behavior and API changes all of a sudden.
And, if we set it to be enabled by default, that would be _all of a sudden_ when somebody upgrades to the next release.

But, 3 people know more than 2! Let's see what @abuijze thinks about this.

Reply:
So, @abuijze, should we default to automatically reroute local queries to local handlers?